src/engine.rs

```diff
@@ -43,11 +44,17 @@ pub fn run(script_path: PathBuf, input_path: PathBuf) -> Result<FunctionRunResul
         let start = Instant::now();
 
         let instance = linker.instantiate(&mut store, &module)?;
+
+        let memory = instance
+            .get_memory(&mut store, "memory")
+            .ok_or(anyhow::format_err!("failed to find `memory` export"))?;
```

Comment:
Is this a standard? Do we need to document this somewhere?

Reply:
Its the way they do it in the docs and in their book: https://docs.rs/wasmtime/0.37.0/wasmtime/struct.Memory.html
https://docs.wasmtime.dev/contributing-architecture.html?highlight=memory#linear-memory

Reply:
I don't see any use of the magic `memory` name in those docs 🤔 In fact the first link uses `mem` in some of their examples

Reply:
Could we potentially use the [exports function](https://docs.rs/wasmtime/0.37.0/wasmtime/struct.Instance.html#method.exports)?

Reply:
I was curious about this as well so took a look. It seems that `memory` is the [standard export name](https://hemingwayapp.com/google.com) when compiling Rust to the `wasm` target. I'm not sure you can change this except if you modified the resulting binary rust gets compiled into. I wouldn't recommend documenting this since it might cause more confusion. Unless the developer is writing a script in pure `WAT` which would be...painful 😆.

A single `wasm` module can [have only one memory](https://webassembly.github.io/spec/core/syntax/modules.html#syntax-mem:~:text=In%20the%20current%20version%20of%20WebAssembly%2C%20at%20most%20one%20memory%20may%20be%20defined%20or%20imported%20in%20a%20single%20module), but many memories is [being proposed](https://github.com/WebAssembly/multi-memory/blob/main/proposals/multi-memory/Overview.md). At that point, I imagine this will need updating.

Reply:
>It seems that memory is the [standard export name](https://rustwasm.github.io/docs/book/reference/js-ffi.html#from-the-rust-side:~:text=In%20addition%20to,exported%20as%20%22memory%22) when compiling Rust to the wasm target.

Being a Rust standard is insufficient. ShopifyVM is a Wasm platform, anything we do must work generally for Wasm modules.

Reply:
Summing up everything they export seems like a reasonable solution,  given that we have access to list of memory export names. I'd rather not document a specific naming limitation if we could remove the limitation instead.

Reply:
This is not a Rust standard, is a standard across compiler infrastructures (LLVM, Binaryen). As Kevin mentioned, you could post-process your Wasm and change the name of the exported memory, not sure what you'd gain by doing so though. Once multiple memories are officially supported by the compiler toolchains I'm pretty sure we'd need to update this. For now this seems a good, standard approach.

Reply:
Also multiple memories is going to introduce some complexities to measure resource usage. I suspect that with multiple memories we'd need to implement the Resource Limiter because not all memories declared in a Wasm module are expected to be exported. So the current approach won't be accurate.

Reply:
>So the current approach won't be accurate.

True, but that's true regardless of whether or not we adopt the `"memory"` name standard. If that's an issue, let's open a separate separate thread about the approach of the PR in general.

In the latest commit Dominic added an approach that sums up all exported memories, which avoids this naming limitation. Is there a reason to re-add the limitation?

Reply:
Yeah, the approach looks good. With the recent changes I don't see the need to reintroduce the "memory" naming standard. I emphasize on multi-memory because when we support it,  for example for module linking, the approach here will need to differ considerably. Just want to make sure that this is on your radar.